sensing/pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -123,10 +126,13 @@ void DistortionCorrectorComponent::onPointCloud(PointCloud2::UniquePtr points_ms
     return;
   }
 
+  Eigen::Matrix4f eigen_base_link_to_sensor;
   tf2::Transform tf2_base_link_to_sensor{};
-  getTransform(points_msg->header.frame_id, base_link_frame_, &tf2_base_link_to_sensor);
+  getTransform(
+    points_msg->header.frame_id, base_link_frame_, &tf2_base_link_to_sensor,
+    eigen_base_link_to_sensor);
 
-  undistortPointCloud(tf2_base_link_to_sensor, *points_msg);
+  undistortPointCloud(tf2_base_link_to_sensor, eigen_base_link_to_sensor, *points_msg);
```

Comment:
@knzo25 
Thanks for reviewing.
I will create a PR soon that refactors all of this redundant stuff. 
It would be better to just review the algorithm concept :) 
I will merge them after both PRs are approved.

Reply:
Ok. Any other that refactor-level changes, looks good to me !